### PR TITLE
eventing set config now sets log tables

### PR DESCRIPTION
**Pull request by KavishShah09**

### CODE REVIEW

File: gateway/modules/eventing/eventing.go [hunk 1]
```diff
@@ -95,7 +97,15 @@ func (m *Module) SetConfig(project string, eventing *config.Eventing) error {
...
-	if eventing.DBType == "" || eventing.Col == "" {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	err := m.schema.SchemaModifyAll(ctx, eventing.DBType, project, map[string]*config.TableRule{eventingLogs: &config.TableRule{Schema: eventSchema}, invocationLogs: &config.TableRule{Schema: invocationSchema}})
+	if err != nil {
+		logrus.Errorf("Could not create tables required for eventing module - %s", err.Error())
+		return err
+	}
+
+	if eventing.DBType == "" {
```
Comment (sharadregoti) [inline gateway/modules/eventing/eventing.go]: check this condition before creating tables

File: gateway/modules/eventing/eventing.go [hunk 2]
```diff
@@ -95,7 +97,15 @@ func (m *Module) SetConfig(project string, eventing *config.Eventing) error {
...
 		return errors.New("invalid eventing config provided")
```
Comment (sharadregoti) [inline gateway/modules/eventing/eventing.go]: after creating tables update the config.
Use syncman.SetModifySchema

File: gateway/modules/eventing/handle_intents.go [hunk 3]
```diff
@@ -86,14 +85,14 @@ func (m *Module) processIntent(eventDoc *model.EventDocument) {
 		if _, err := m.crud.Read(ctx, createEvent.DBType, m.project, createEvent.Col, readRequest); err != nil {
 
 			// Mark event as cancelled if it document doesn't exist
-			if err := m.crud.InternalUpdate(ctx, m.config.DBType, m.project, m.config.Col, m.generateCancelEventRequest(eventID)); err != nil {
+			if err := m.crud.InternalUpdate(ctx, m.config.DBType, m.project, eventingLogs, m.generateCancelEventRequest(eventID)); err != nil {
 				log.Println("Eventing: Couldn't cancel intent -", err)
```
Comment (sharadregoti) [inline gateway/modules/eventing/handle_intents.go]: change log to logrus everywhere if this file

File: gateway/modules/eventing/handle_staged.go [hunk 4]
```diff
@@ -130,7 +128,7 @@ func (m *Module) processStagedEvent(eventDoc *model.EventDocument) {
 		return
 	}
 
-	if err := m.crud.InternalUpdate(context.Background(), m.config.DBType, m.project, m.config.Col, m.generateFailedEventRequest(eventDoc.ID, "Max retires limit reached")); err != nil {
+	if err := m.crud.InternalUpdate(context.Background(), m.config.DBType, m.project, eventingLogs, m.generateFailedEventRequest(eventDoc.ID, "Max retires limit reached")); err != nil {
 		log.Println("Eventing staged event handler could not update event doc:", err)
```
Comment (sharadregoti) [inline gateway/modules/eventing/handle_staged.go]: here also

**Comment by KavishShah09** [comment]
Tested locally
Space cloud creates tables on adding a database

**Comment by KavishShah09** [comment]
Fixes #817

**Comment by sharadregoti** [comment]
where did u call syncman.SetModifyschema for updating schemas in config ?

**Comment by KavishShah09** [comment]
@sharadregoti that creates a deadlock so it's not possible

**Pull request merged**